controller/User/getWatchList.js

```diff
@@ -1,50 +1,57 @@
 const models = require('../../models');
 var jwt = require('jsonwebtoken')
+/** @description This functions gives list of all movies present in the watchlist
+ * @param {object} req - Request object will be null
+ * @param {object} res -  Reponse object with all the movies list  if success or error message if there is an error.
+ * @param {requestCallback} next - The callback that calls the error handling middleware.
+ *  @returns {Promise}
+*/
 const getWatchList = async (req, res, next) => {
     try {
-        
+
         const token = req.headers['access-token']
         const payload = jwt.decode(token)
         const user = await models.User.findOne({
             where: {
-                userName: payload.userName
+                userName: req.params.userName
             }
         })
-        const list=await models.Userlist.findAll({
-            where:{
-                userId:user.id,
-                listType:"watch"
-                
+        const list = await models.Userlist.findAll({
+            where: {
+                userId: user.id,
+                listType: "watch"
+
             }
         })
 
-        var movieslist=[]
+        var movieslist = []
         obj = [...JSON.parse(JSON.stringify(list, null, 4))];
-        for(i=0;i<obj.length;i++){
-            var movie=await models.Movie.findOne({
-                where:{id:obj[i].movieId} ,
-                include:[{
-                    model:models.MoviePerson,
-                    required:true,
-                    include:[{
-                        model:models.Person,
-                        required:true},
-                        {model:models.MoviePersonRole}
-                       
-                        
+        for (i = 0; i < obj.length; i++) {
```

Comment:
Same here. Use map

Reply:
I have changed all for loops to map but only in one api i am not able to because if i use that  i am not getting correct output